src/main/java/com/phonebookservice/model/Contact.java

```diff
@@ -0,0 +1,62 @@
+package main.java.com.phonebookservice.model;
+
+import java.util.List;
+
+public class Contact implements Model {
+    private String firstName;
+    private String lastName;
+    private String city;
+    private List<Address> addresses;
+    private List<PhoneNumber> phoneNumbers;
+    private List<SocialProfile> socialProfiles;
+
+    /**
+     * contact constructor.
+     *
+     * @param firstNameValue the first name.
+     */
+    public Contact(final String firstNameValue) {
+        this.firstName = firstNameValue;
+    }
+
+    /**
+     * contact constructor.
+     *
+     * @param firstName            the first name.
+     * @param lastName             the last name.
+     * @param city                 the city.
+     * @param listOfAddress        the addresses.
+     * @param listOfPhoneNumbers   the phone numbers.
+     * @param listOfSocialProfiles the social profiles.
+     */
+    public Contact(final String firstName, final String lastName,
+            final String city, final List<Address> listOfAddress,
+            final List<PhoneNumber> listOfPhoneNumbers,
+            final List<SocialProfile> listOfSocialProfiles) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.city = city;
+        this.addresses = listOfAddress;
+        this.phoneNumbers = listOfPhoneNumbers;
+        this.socialProfiles = listOfSocialProfiles;
+    }
```

Comment:
Based on above comments :- 
1- Constructor should contain all mandatory variable (members) e.g controller has database in constructor because there is no controller without database. In Contact, what are the mandatory variables (members) ? 

2- Based on our previous discussion, we don't add setter methods if we don't want to change on those member after creating the object, what is the name of this design pattern? 

3- If you would like to have optional members, e,g (addresses , social profiles), you will use setter method after creating the object, right ? Answer is yes 
```
Contact contact = new Contact(lastName, firstName);
contact.setAddresses(addressList);
contact.setProfile(profile);
```

You can avoid the above lines and have `builder design pattern` like the following 
```
Contact contact = ContactBuilder.withFirstName("").withLastName("").withaddress("").build();
```

read about builder design pattern, and answer to me, when should we use builder design pattern ?

Reply:
```
Contact contact = Contact.builder().withFirstName("").withLastName("").withaddress("").build();
```

Reply:
```
class ContactBuilder {
String firstName;
String lastName;

ContactBuilder withLastName(String lastName) {
      this.lastName = lastName;
      return this;

}
}

```